test/test_fiat_p256.ml

```diff
@@ -38,6 +38,227 @@ let whole_key_exchange =
   in
   [(test_name, `Quick, test_fun)]
 
+let key_pair_of_hex h = Fiat_p256.gen_key ~rng:(fun _ -> Hex.to_cstruct h)
+
+let scalar_of_hex h = fst (key_pair_of_hex h)
+
+let pp_hex_le fmt cs =
+  let n = Cstruct.len cs in
+  for i = n - 1 downto 0 do
+    let byte = Cstruct.get_uint8 cs i in
+    Format.fprintf fmt "%02x" byte
+  done
+
+let pp_result ppf = function
+  | Ok cs -> pp_hex_le ppf cs
+  | Error e -> Format.fprintf ppf "%a" Fiat_p256.pp_error e
+
+let key_exchange =
+  let test ~name d p ~expected =
+    ( name
+    , `Quick
+    , fun () ->
+        Fiat_p256.key_exchange d p
+        |> Format.asprintf "%a" pp_result
+        |> Alcotest.check Alcotest.string __LOC__ expected )
+  in
+  let d_a, p_a =
+    key_pair_of_hex
+      (`Hex
+        "200102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f")
+  in
+  let d_b, p_b =
+    key_pair_of_hex
+      (`Hex
+        "000102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f")
+  in
+  [ test ~name:"b*A" d_b p_a
+      ~expected:
+        "2e3e4065a62a7f425aaf8aae3d158f367c733300b5002e0b62f4bc6260789e1b"
+  ; test ~name:"a*B" d_a p_b
+      ~expected:
+        "2e3e4065a62a7f425aaf8aae3d158f367c733300b5002e0b62f4bc6260789e1b"
+  ; test ~name:"a*A" d_a p_a
+      ~expected:
+        "2ea4e810837da217a5bfd05f01d12459eeda830b6e0dec7f8afa425c5b55c507"
+  ; test ~name:"b*B" d_b p_b
+      ~expected:
+        "a7666bcc3818472194460f7df22d80a5886da0e1679eac930175ce1ff733c7ca" ]
+
+let scalar_mult =
+  let test ~scalar ~point ~expected =
+    let scalar = scalar_of_hex scalar in
+    let point = Hex.to_cstruct point in
+    ( "Scalar mult"
```

Comment:
I tend to use more expressive name for each test case. Usually it looks like `function_name: test_case_name` as it makes the output of `alcotest` easier to interpret as you know straight away which test/function needs fixing.

I also usually pass that same string to `Alcotest.check` for the same reasons.

What do you think about using that same convention here?

*Food for thought: I wonder if we could write a ppx to deal with all that `Alcotest` boiler plate*

Reply:
Those were extracted from logs, so there are not all meaningful and I can't think of a great naming scheme. I added a `mapi` to add unique names, how does that look?

Reply:
How about passing the index as a name explicitly so that we don't have to count?

Something like:
```
[ test ~name:"0" ...
; test ~name:"1" ...
```

That way we can aslo strip the extra `run` function.

Reply:
done!